### PR TITLE
Wip scrub repair

### PR DESCRIPTION
**Pull request by ukernel**

### CODE REVIEW

File: tasks/cephfs/test_scrub_checks.py [hunk 1]
```diff
@@ -141,14 +134,65 @@ def json_validator(json_out, rc, element, expected_value):
...
+                      pool=self.fs.get_metadata_pool_name())
+
+        self.fs.mds_fail_restart()
+        self.fs.wait_for_daemons()
+
+        self.mount_a.mount()
+        self.mount_a.wait_until_mounted()
+
+        # fragstat indicates the directory is not empty, rmdir should fail
+        try:
```
Comment (jcsp) [inline tasks/cephfs/test_scrub_checks.py]: A neater way to assert that something should raise an exception is with "self.assertRaises" (see uses elsewhere in tests)

**Comment by gregsfortytwo** [comment]
I think it's just this test is broken, but it needs fixing (unless I merged it wrong?)
http://pulpito.ovh.sepia.ceph.com:8081/gregf-2015-12-11_09:34:05-fs-greg-fs-testing---basic-openstack/37437/
http://pulpito.ovh.sepia.ceph.com:8081/gregf-2015-12-11_09:34:05-fs-greg-fs-testing---basic-openstack/37475/

**Comment by ukernel** [comment]
On Tue, Dec 15, 2015 at 11:30 AM, Gregory Farnum <notifications@github.com>
wrote:

> I think it's just this test is broken, but it needs fixing (unless I
> merged it wrong?)
>
> http://pulpito.ovh.sepia.ceph.com:8081/gregf-2015-12-11_09:34:05-fs-greg-fs-testing---basic-openstack/37437/
>
> http://pulpito.ovh.sepia.ceph.com:8081/gregf-2015-12-11_09:34:05-fs-greg-fs-testing---basic-openstack/37475/
>
> —
>

I think Greg didn't merge the wip-scrub-repair branch into greg-fs-testing
branch.

**Comment by gregsfortytwo** [comment]
I'm afraid I did merge it in, although there was a conflict so I might have messed up the logic (but I don't think so). See https://github.com/ceph/ceph-qa-suite/commit/55a00b6af8100447db5dffbf67daee1258dba7bf for the history — it was definitely on this commit or some other things would have behaved very differently than they did.

**Comment by ukernel** [comment]
On Fri, Dec 18, 2015 at 4:05 PM, Gregory Farnum <notifications@github.com>
wrote:

> I'm afraid I did merge it in, although there was a conflict so I might
> have messed up the logic (but I don't think so). See 55a00b6
> <https://github.com/ceph/ceph-qa-suite/commit/55a00b6af8100447db5dffbf67daee1258dba7bf>
> for the history — it was definitely on this commit or some other things
> would have behaved very differently than they did.
>
>
I mean you didn't merge wip-scrub-repair branch into master branch of ceph
repository.

**Comment by gregsfortytwo** [comment]
...yep, okay, not sure how that mismatch happened.

**Comment by jcsp** [comment]
This had conflicts with master, I pushed a rebase of it.

File: tasks/cephfs/test_scrub_checks.py [hunk 2]
```diff
@@ -130,16 +123,64 @@ def json_validator(json_out, rc, element, expected_value):
...
+
+        # fragstat indicates the directory is not empty, rmdir should fail
+        with self.assertRaises(CommandFailedError) as ar:
+            self.mount_a.run_shell(["sudo", "rmdir", test_dir])
+        self.assertEqual(ar.exception.exitstatus, 1)
+
+        self.asok_command(mds_rank, "scrub_path /{0} repair".format(test_dir),
+                          lambda j, r: self.json_validator(j, r, "return_code", 0))
+
+	# wait a few second for background repair
```
Comment (jcsp) [inline tasks/cephfs/test_scrub_checks.py]: The whitespace is weird here (indentation switches from spaces to tabs)

**Pull request merged**